apps/dev/src/commands/feeds/read/adfs.ts

```diff
@@ -9,9 +9,43 @@ import {
 import { renderTui, drawTable } from '@blocksense/base-utils/tty';
 import { listEvmNetworks, readEvmDeployment } from '@blocksense/config-types';
 import { AggregatedDataFeedStoreConsumer } from '@blocksense/contracts/viem';
+import { formatTimestamp } from '../../utils';
 
 const availableNetworks = await listEvmNetworks();
 
+export const parseADFSData = (hexData: string) => {
+  const cleanHex = hexData.startsWith('0x') ? hexData.slice(2) : hexData;
+  const valuePart = cleanHex.slice(0, 48);
+  const valueHex = '0x' + valuePart.padEnd(64, '0');
+  let value: bigint = 0n;
+  try {
+    value = BigInt(valueHex);
+  } catch {
+    // ignore
+  }
+  let timestamp: bigint | null = null;
+  let timestampHex: string | null = null;
+  if (cleanHex.length >= 64) {
+    timestampHex = '0x' + cleanHex.slice(48, 64);
+    try {
+      timestamp = BigInt(timestampHex);
+    } catch {
+      timestamp = null;
+    }
+  }
+  const formattedTimestamp =
+    timestamp !== null ? formatTimestamp(timestamp) : null;
+
+  return {
+    rawHex: hexData,
+    value: value.toString(),
+    valueHex,
+    timestamp: timestamp ? timestamp.toString() : null,
+    timestampHex,
+    formattedTimestamp,
+  } as const;
```

Comment:
In comments bellow I suggest we do not visualize hex representation of `value` and `timestamp` as it is present in the `rawHex` already. So you can simplify the return object.
```suggestion
  return {
    rawHex: hexData,
    value: value.toString(),
    timestamp: formattedTimestamp,
  };
```